client.go

```diff
@@ -18,10 +20,22 @@ type ldapClient interface {
 	Execute(conf *client.Config, entries []*ldif.Entry, continueOnError bool) (err error)
 }
 
-func NewClient() *Client {
-	return &Client{
+func NewClient(conf *logical.BackendConfig) *Client {
+	client := &Client{
 		ldap: client.New(),
 	}
+
+	logger := conf.Logger
+
+	// This can only happen in a test where BackendConfig is created
+	// manually without a logger.
+	if logger == nil {
+		logger = hclog.NewNullLogger()
+	}
+
+	client.ldap.LDAP.Logger = logger
```

Comment:
Would it be cleaner to pass in a logger on the client itself like so `client.New(logger)`?